File: test/fixtures/complexity/complex.ts
// rank: ciclomática hand-counted = 6 (1 base + for + if + && + else-if + ternary)
export function rank(items: number[], strict: boolean): number {
  let score = 0;
  for (const it of items) {
    if (it > 0 && strict) {
      score += it;
    } else if (it < -5) {
      score -= 1;
    }
  }
  return score > 0 ? score : 0;
}

// arrow simples: ciclomática = 1
export const noop = (): number => 42;

File: test/fixtures/complexity/Complex.java
package com.acme;

public class Complex {
    // classify: ciclomática hand-counted = 8 (1 base + if + for + if + && + else-if + while + ||)
    // maxNesting = 3 (if -> for -> if)
    public int classify(int n, boolean flag) {
        int result = 0;
        if (n > 10) {
            for (int i = 0; i < n; i++) {
                if (i % 2 == 0 && flag) {
                    result += i;
                }
            }
        } else if (n < 0) {
            result = -1;
        }
        while (result > 100 || flag) {
            result -= 10;
        }
        return result;
    }

    // simple: ciclomática = 1, sem aninhamento
    public void simple() {
        System.out.println("hi");
    }
}
